angular/projects/admin-nrpti/src/app/news/news-add-edit/news-add-edit.component.ts

```diff
@@ -128,19 +128,20 @@ export class NewsAddEditComponent implements OnInit, OnDestroy {
 
     if (!this.isEditing) {
       // Add the news item.
-      this.factoryService.createNews(newsItem).subscribe(async res => {
-        // TODO: Need to check for res.length, as the return is typed object, but it's actually an array.
-        if (!res /*|| !res.length*/ || !res[0] || !res[0].length || !res[0][0]) {
-          alert('Failed to create News Item.');
-        } else {
-          this.router.navigate(['news', this.record.system, res[0][0].object._id, 'detail']);
-        }
-      });
+      const res = await this.factoryService.createNews(newsItem);
+      if (!res || res.insertedCount !== 1) {
```

Comment:
Should be:
`if (!res || res.ok !== 1) {`

Or, if you're sending back a response from the API, double check you're sending back the custom response and not the mongo response.

Reply:
3a32d6f8c9e96a7ef201e56a13995a6357f2498e